jobrunner/create_or_update_jobs.py

```diff
@@ -337,10 +341,17 @@ def create_failed_job(job_request, exception):
         error = None
     else:
         state = State.FAILED
-        code = StatusCode.INTERNAL_ERROR
-        status_message = f"{type(exception).__name__}: {exception}"
+
+        if isinstance(exception, StaleCodelistError):
+            code = StatusCode.STALE_CODELISTS
+            status_message = str(exception)
+        else:
+            code = StatusCode.INTERNAL_ERROR
+            status_message = f"{type(exception).__name__}: {exception}"
+
```

Comment:
Might be slightly more readable to collapse this down to:
```python
if isinstance(exception, NothingToDoError):
    ...
elif isinstance(exception, StaleCodelistError):
    ...
else:
    ...
```

It would mean repeating `state = State.FAILED` on two branches, but I think the control flow is easier to read than nesting it.

Reply:
It's a bit more than `state`, but I've made it work.